src/vma/dev/qp_mgr.cpp

```diff
@@ -123,10 +124,53 @@ qp_mgr::~qp_mgr()
 	munmap(m_rq_wqe_idx_to_wrid, m_rx_num_wr * sizeof(*m_rq_wqe_idx_to_wrid));
 	munmap(m_sq_wqe_idx_to_wrid, m_tx_num_wr * sizeof(*m_sq_wqe_idx_to_wrid));
 #endif // DEFINED_VMAPOLL
+
+#ifndef DEFINED_VMAPOLL
+	delete[] m_rq_wqe_idx_to_wrid;
+#else
+	free(m_rq_wqe_idx_to_wrid);
+#endif
 	qp_logdbg("Rx buffer poll: %d free global buffers available", g_buffer_pool_rx->get_free_count());
 	qp_logdbg("delete done");
 }
 
+static inline uint32_t align32pow2(uint32_t x)
+{
+	x--;
+	x |= x >> 1;
+	x |= x >> 2;
+	x |= x >> 4;
+	x |= x >> 8;
+	x |= x >> 16;
+
+	return x + 1;
+}
+
+cq_mgr*	qp_mgr::init_rx_cq_mgr(struct ibv_comp_channel* p_rx_comp_event_channel)
+{
+	 return new cq_mgr(m_p_ring, m_p_ib_ctx_handler, m_rx_num_wr, p_rx_comp_event_channel, true);
+}
+
+#if !defined(DEFINED_VMAPOLL) && defined(HAVE_INFINIBAND_MLX5_HW_H)
```

Comment:
Better to have something like:
```
#if !defined(DEFINED_VMAPOLL) && defined(HAVE_INFINIBAND_MLX5_HW_H)
#define DEFINE_VMA_PRM_MLX5_SUPPORT
#else
#undef DEFINE_VMA_PRM_MLX5_SUPPORT
#endif
```
and use:
```
#if defined(DEFINE_VMA_PRM_MLX5_SUPPORT)
        ...
#endif
```

This is best handled in _configure.ac_